`linalg_lu_factor_ex`: change to new `MetaBase` API.

ysiraichi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #78039
* #78038
* #77540
* __->__ #77539
* #76098
* #76097

This PR modifies `linalg_lu_factor_ex` to the new `MetaBase` API. Instead of checking the
output strides manually, it now uses the appropriate `set_output_strided` and
`set_output_contiguous`.

Add for lu

ysiraichi committed:
`linalg_lu_factor_ex`: change to new `MetaBase` API.

This PR modifies `linalg_lu_factor_ex` to the new `MetaBase` API. Instead of checking the
output strides manually, it now uses the appropriate `set_output_strided` and
`set_output_contiguous`.

Add for lu

[ghstack-poisoned]

Lezcano commented on test/test_linalg.py:
```diff
@@ -5211,6 +5211,39 @@ def test_householder_product_errors_and_warnings(self, device):
+        fn(A, out=out_)
+
+        for i in range(len(out)):
+            # Check whether the results are equal for the 'out' overload.
+            self.assertEqual(out[i], out_[i])
+            # Check whether the strides were preserved after the call.
+            self.assertEqual(strides_[i], out_[i].stride())
+            # Check whether the original strides are different from the generated ones.
+            self.assertNotEqual(out[i].stride(), out_[i].stride())
```
I believe these things are already tested in the `OpInfo` tests in
https://github.com/pytorch/pytorch/blob/c83f8ee46ab057a8bce7ed7ba7ceccba5c388d8f/test/test_ops.py#L676-L684

ysiraichi replied:
Ah, you are right! Thanks for pointing that out.

lezcano reviewed (commented):
This is fantastic! Thank you so much!

The only thing here is that I believe that the whole test is not necessary, as these things are already tested in the `OpInfo` for this operation.

ysiraichi committed:
Rebase. on "`linalg_lu_factor_ex`: change to new `MetaBase` API."


This PR modifies `linalg_lu_factor_ex` to the new `MetaBase` API. Instead of checking the
output strides manually, it now uses the appropriate `set_output_strided` and
`set_output_contiguous`.

Add for lu

[ghstack-poisoned]

ysiraichi committed:
Remove redundant test. on "`linalg_lu_factor_ex`: change to new `MetaBase` API."


This PR modifies `linalg_lu_factor_ex` to the new `MetaBase` API. Instead of checking the
output strides manually, it now uses the appropriate `set_output_strided` and
`set_output_contiguous`.

Add for lu

[ghstack-poisoned]

ysiraichi committed:
Fix new 'set_output' that wasn't replaced. on "`linalg_lu_factor_ex`: change to new `MetaBase` API."


This PR modifies `linalg_lu_factor_ex` to the new `MetaBase` API. Instead of checking the
output strides manually, it now uses the appropriate `set_output_strided` and
`set_output_contiguous`.

Add for lu

[ghstack-poisoned]

lezcano reviewed (approved):
This is just great :D

ezyang commented:
@pytorchbot merge this

ezyang commented:
@pytorchbot merge this
